tmt/steps/prepare/install.py

```diff
@@ -163,8 +163,22 @@ def install_from_url(self) -> None:
     def install_debuginfo(self) -> None:
         """ Default base install method for debuginfo packages """
 
+    def make_cache(self) -> None:
+        """ Default base method to generate metadata cache"""
+
     def install(self) -> None:
         """ Perform the actual installation """
+        try:
+            self.perform_installation()
+        except Exception:
+            # Refresh cache in case of recent but not updated change to repodata
+            if str(self.guest.package_manager) in ["dnf", "dnf5", "yum"]:
+                self.make_cache()
+            # Retry the installation
+            self.perform_installation()
+
+    def perform_installation(self) -> None:
```

Comment:
I'd recommend a different name - `_install`, or `_do_install` or something similar. It should be less public, i.e. starting with `_` would be wise, and it should immediately resemble its "parent", `install`. `perform_installation` is way too public API-ish, and does not look like it's related to `install()`, quite contrary, it looks like a standalone operation.

Reply:
Fixed.